src/ManageCourses.UcasCourseImporter/importer/UcasDataMigrator.cs

```diff
@@ -209,12 +221,10 @@ private Provider ToProvider(UcasInstitution x)
             };
         }
 
-        private Provider UpsertProvider(Provider newValues)
+        private Provider UpsertProvider(Provider newValues, IReadOnlyDictionary<string, Provider> providerCache)
```

Comment:
Readonly collection? Don't we need to add it that collection?

Reply:
it's done outside that method call